app/models/topic.rb

```diff
@@ -1802,7 +1802,12 @@ def rate_limit_topic_invitation(invited_by)
   end
 
   def cannot_permanently_delete_reason(user)
-    if self.posts_count > 0
+    deleted_posts_count = Post.with_deleted
+      .where(topic_id: self.id)
+      .where(post_type: [Post.types[:regular], Post.types[:moderator_action], Post.types[:whisper]])
+      .count
+
+    if posts_count > 0 || deleted_posts_count > 1
```

Comment:
Why the `> 1` for `deleted_posts_count`?

Reply:
The current post is included in the count for `deleted_posts_count`. If there is another soft-deleted post in the topic, then the first post cannot be deleted.